src/sentry/web/client_config.py

```diff
@@ -139,13 +144,26 @@ def get_client_config(request=None):
 
     public_dsn = _get_public_dsn()
 
-    last_organization = session["activeorg"] if session and "activeorg" in session else None
-    try:
-        Organization.objects.get_from_cache(slug=last_organization)
-    except Organization.DoesNotExist:
-        last_organization = None
-        if session and "activeorg" in session:
-            del session["activeorg"]
+    last_org_slug = session["activeorg"] if session and "activeorg" in session else None
+    if not last_org_slug:
+        last_org_slug = None
+        _delete_activeorg(session)
+    else:
+        try:
+            last_org = Organization.objects.get_from_cache(slug=last_org_slug)
+            if user is not None and not isinstance(user, AnonymousUser):
+                has_membership = OrganizationMember.objects.filter(
+                    user=user, organization=last_org
+                ).exists()
+                if not has_membership:
+                    last_org_slug = None
+                    _delete_activeorg(session)
+            else:
+                last_org_slug = None
+                _delete_activeorg(session)
+        except Organization.DoesNotExist:
+            last_org_slug = None
+            _delete_activeorg(session)
```

Comment:
Should you separate out the slug access validation from overwriting the session[activeorg] value? You could have something like

```python
last_org_slug = _validate_activeorg(request)
if not last_org_slug:
   _delete_activeorg(session)
```

You might also be able to use `request.access.member` to access the current user's membership on the organization instead of having to do another query for it.

Reply:
It doesn't seem like `access` is always present in `request`. I opted to use `get_cached_organization_member()` instead.